api/pkg/transformer/types/encoder/cyclical_encoder.go

```diff
@@ -0,0 +1,312 @@
+package encoder
+
+import (
+	"fmt"
+	"github.com/gojek/merlin/pkg/transformer/spec"
+	"github.com/gojek/merlin/pkg/transformer/types/converter"
+	"math"
+	"time"
+)
+
+const (
+	MinInSec  = 60
+	HourInSec = 3600
+	DayInSec  = 86400
+	WeekInSec = 604800
+
+	DaysInSec31 = 2678400
+	DaysInSec30 = 2592000
+	DaysInSec29 = 2505600
+	DaysInSec28 = 2419200
+
+	Q1InSec     = 7776000
+	Q1LeapInSec = 7862400
+	Q2InSec     = 7862400
+	Q3InSec     = 7948800
+	Q4InSec     = 7948800
+
+	H1InSec     = 15638400
+	H1LeapInSec = 15724800
+	H2InSec     = 15897600
+
+	YearInSec     = 31536000
+	LeapYearInSec = 31622400
+
+	completeAngle = 2 * math.Pi
+
+	//Unit angles for variable periods
+	UnitDaysInSec31 = completeAngle / DaysInSec31
+	UnitDaysInSec30 = completeAngle / DaysInSec30
+	UnitDaysInSec29 = completeAngle / DaysInSec29
+	UnitDaysInSec28 = completeAngle / DaysInSec28
+
+	UnitQ1InSec     = completeAngle / Q1InSec
+	UnitQ1LeapInSec = completeAngle / Q1LeapInSec
+	UnitQ2InSec     = completeAngle / Q2InSec
+	UnitQ3InSec     = completeAngle / Q3InSec
+	UnitQ4InSec     = completeAngle / Q4InSec
+
+	UnitH1InSec     = completeAngle / H1InSec
+	UnitH1LeapInSec = completeAngle / H1LeapInSec
+	UnitH2InSec     = completeAngle / H2InSec
+
+	UnitYearInSec     = completeAngle / YearInSec
+	UnitLeapYearInSec = completeAngle / LeapYearInSec
+)
+
+//Unit angles for variable periods for each month
+var MonthInSec = map[int]float64{
+	1:  UnitDaysInSec31,
+	2:  UnitDaysInSec28,
+	3:  UnitDaysInSec31,
+	4:  UnitDaysInSec30,
+	5:  UnitDaysInSec31,
+	6:  UnitDaysInSec30,
+	7:  UnitDaysInSec31,
+	8:  UnitDaysInSec31,
+	9:  UnitDaysInSec30,
+	10: UnitDaysInSec31,
+	11: UnitDaysInSec30,
+	12: UnitDaysInSec31,
+	13: UnitDaysInSec29, //Leap year Feb
+}
+
+type CyclicalEncoder struct {
+	Period interface{}
```

Comment:
Why is `Period` an `interface{}`?

Reply:
So that I can accept a "nil" when we use min max instead... hmm... ok, maybe I can set that to 0

Reply:
oh wait. should be enum.